tests/error/misc_errors/return_flag_callable.err

```diff
@@ -0,0 +1,6 @@
+Guppy compilation failed. Error in file $FILE:14
+
+12:    @guppy.declare(module)
+13:    def foo(f: Callable[[], qubit @inout]) -> None: ...
+                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
+GuppyError: `@` type annotations are not allowed in this position
```

Comment:
The broad error location is not ideal, but improving this requires a bigger refactor